crypto/rand/sysrand.c

```diff
@@ -88,21 +86,16 @@ int GFp_sysrand_chunk(void *out, size_t requested) {
 #endif
 #endif
 
-
-/* http://man7.org/linux/man-pages/man2/getrandom.2.html: "Calling
- * getrandom() to read /dev/urandom for small values (<= 256) of buflen is
- * the preferred mode of usage." */
-const size_t GFp_sysrand_chunk_max_len = 256;
-
 int GFp_sysrand_chunk(void *out, size_t requested) {
-  assert(requested <= GFp_sysrand_chunk_max_len);
-  if (syscall(SYS_getrandom, out, requested, 0u) < 0) {
-    if (errno == ENOSYS) {
-      return -1;
+  int r = syscall(SYS_getrandom, out, requested, 0u);
+  if (r < 0) {
+    // EINTR and EAGAIN are normal, and we can try again. Other error codes are fatal.
+    if (errno == EINTR || errno == EAGAIN) {
```

Comment:
We don't need to handle `EAGAIN` since we're not using the non-blocking flag, right? We can just treat it like other errors, I think.